app/models/spree/order_decorator.rb

```diff
@@ -0,0 +1,17 @@
+module Spree
+  module OrderDecorator
+    def self.prepended(base)
+      return if base.checkout_steps.has_key?(:payment_confirm)
+
+      base.insert_checkout_step(
+        :payment_confirm,
+        before: :confirm,
```

Comment:
@szymoniwacz `confirm` step is optional, can be turned off - maybe it would be better to put that after `payment` step?

Reply:
You are right, thanks for spotting that!